src/test/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSparkOptimizedIntegrationTest.java

```diff
@@ -0,0 +1,192 @@
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class BaseRecalibratorSparkOptimizedIntegrationTest extends CommandLineProgramTest {
+
+    private final static String THIS_TEST_FOLDER = "org/broadinstitute/hellbender/tools/BQSR/";
+
+    private static class BQSRTest {
+        final String referenceURL;
+        final String bam;
+        final String knownSites;
+        final String args;
+        final String expectedFileName;
+
+        private BQSRTest(String referenceURL, String bam, String knownSites, String args, String expectedFileName) {
+            this.referenceURL = referenceURL;
+            this.bam = bam;
+            this.knownSites = knownSites;
+            this.args = args;
+            this.expectedFileName = expectedFileName;
+        }
+
+        public String getCommandLineNoApiKey() {
+            return  " -R " + referenceURL +
+                    " -I " + bam +
+                    " " + args +
+                    (knownSites.isEmpty() ? "": " -knownSites " + knownSites) +
+                    " -O %s" +
+                    " -sortAllCols";
+        }
+
+        public String getCommandLine() {
+            return  getCommandLineNoApiKey() +
+                    " --apiKey " + getGCPTestApiKey();
+        }
+
+        @Override
+        public String toString() {
+            return String.format("BQSR(bam='%s', args='%s')", bam, args);
+        }
+    }
+
+    private String getResourceDir(){
+        return getTestDataDir() + "/" + "BQSR" + "/";
+    }
+
+    private String getCloudInputs() {
+        return getGCPTestInputPath() + THIS_TEST_FOLDER;
+    }
+
+    @DataProvider(name = "BQSRTest")
+    public Object[][] createBQSRTestData() {
+        final String localResources =  getResourceDir();
+        final String hg19Ref = "EMWV_ZfLxrDY-wE";
+        final String GRCh37Ref = ReferenceAPISource.URL_PREFIX + ReferenceAPISource.GRCH37_REF_ID;
+        final String HiSeqBam = localResources + "CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.bam";
+        final String dbSNPb37 =  getResourceDir() + "dbsnp_132.b37.excluding_sites_after_129.chr17_69k_70k.vcf";
+        final String moreSites = getResourceDir() + "bqsr.fakeSitesForTesting.b37.chr17.vcf"; //for testing 2 input files
+        final String GRCh37RefLocal = b37_reference_20_21;
+
+        return new Object[][]{
+                // local computation and files (except for the reference)
+                {new BQSRTest(GRCh37Ref, HiSeqBam, dbSNPb37, "", localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.recal.txt")},
+
+                // Currently disabled because BaseRecalibratorSpark can't handle more than 1 knownSites file.
+                // Re-enable once this is fixed.
+                //{new BQSRTest(GRCh37Ref, HiSeqBam, dbSNPb37, "-knownSites " + moreSites, localResources + "expected.CEUTrio.HiSeq.WGS.b37.ch20.1m-1m1k.NA12878.2inputs.recal.txt")},
```

Comment:
This restriction does not exist in `BaseRecalibratorSparkOptimized`, so this test case should be uncommented.